pkg/login/social/social.go

```diff
@@ -225,3 +233,92 @@ var GetOAuthProviders = func(cfg *setting.Cfg) map[string]bool {
 
 	return result
 }
+
+var GetOAuthHttpClient = func(name string) (*http.Client, error) {
+	name = strings.TrimPrefix(name, "oauth_")
+
+	// handle call back
+	tr := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: setting.OAuthService.OAuthInfos[name].TlsSkipVerify,
+		},
+	}
```

Comment:
```suggestion
	info, ok := setting.OAuthService.OAuthInfos[name]
	if !ok {
		return nil, fmt.Errorf("Could not find %s in OAuth Settings", name)
	}
	
	// handle call back
	tr := &http.Transport{
		Proxy: http.ProxyFromEnvironment,
		TLSClientConfig: &tls.Config{
			InsecureSkipVerify: info.TlsSkipVerify,
		},
	}
```

If `name` is not in the `OAuthInfos` map, trying to access `TlsSkipVerify` will panic.